libzecale/tests/circuits/bls12_377_pairing_test.cpp

```diff
@@ -368,6 +368,150 @@ TEST(BLS12_377_PairingTest, MillerLoopGadgetTest)
     ASSERT_TRUE(snark::verify(primary_input, proof, keypair.vk));
 }
 
+TEST(BLS12_377_PairingTest, FinalExpFirstPart)
+{
+    using FieldT = libff::Fr<wpp>;
+    using FqkT = libff::Fqk<npp>;
+    using Fq2T = typename FqkT::my_Fp2;
+    using Fq6T = typename FqkT::my_Fp6;
+
+    // Native calculation
+    const FqkT a(
+        Fq6T(
+            Fq2T(FieldT("1"), FieldT("2")),
+            Fq2T(FieldT("3"), FieldT("4")),
+            Fq2T(FieldT("5"), FieldT("6"))),
+        Fq6T(
+            Fq2T(FieldT("21"), FieldT("22")),
+            Fq2T(FieldT("23"), FieldT("24")),
+            Fq2T(FieldT("25"), FieldT("26"))));
+    const FqkT final_exp_first_part =
+        bls12_377_final_exponentiation_first_chunk(a);
+
+    // Circuit with final exponentiation first part gadget
+    libsnark::protoboard<FieldT> pb;
+    libzecale::Fp12_2over3over2_variable<FqkT> a_var(pb, "a");
+    libzecale::Fp12_2over3over2_variable<FqkT> final_exp_first_part_var(
+        pb, "final_exp_first_part");
+    const size_t num_primary_inputs = pb.num_inputs();
+    pb.set_input_sizes(num_primary_inputs);
+    libzecale::bls12_377_final_exp_first_part_gadget<wpp>
+        final_exp_first_part_gadget(
+            pb, a_var, final_exp_first_part_var, "final exp first part");
```

Comment:
space is the annotation separator. `final exp first part` -> `compute_final_exp_first_part` (the variable `final_exp_first_part_var` already as the annotation `"final_exp_first_part"`)

Reply:
chaged

Reply:
This may be another one to rename, i.e. we may want to rename `final_exp_first_part_gadget` as `compute_final_exp_first_part` - which would be in line with the `compute_v1/0` names for the gadgets used in Karatsuba and would also be in line with the annotation used here.
I'll link this comment to the associated ticket on annotations.